src/containers/EventForm.js

```diff
@@ -34,14 +39,32 @@ export const withCreateEvent = compose(
   })
 );
 
+export const withDelete = compose(
+  getContext({ firebase: PropTypes.object }),
+  withHandlers({
+    onConfirm: ({ item, onCancel }) => () => {
+      // TODO sent to firebase / call EventCRUD prop
+      console.log("DELETE EVENT", item);
+      onCancel(); // close modal after delete
+    }
+  })
+);
+
 export const withEventActionState = compose(
   connect(selectAuth),
   withState("createEventVisible", "setCreateEventVisibility", false),
   withState("editEvent", "setEditEvent", null),
+  withState("confirmDeleteEvent", "setConfirmDelete", null),
   withHandlers({
     toggleCreateEvent: ({
       createEventVisible,
       setCreateEventVisibility
-    }) => () => setCreateEventVisibility(!createEventVisible)
+    }) => () => setCreateEventVisibility(!createEventVisible),
+    toggleConfirmDeleteModal: ({
+      confirmDeleteEvent,
+      setConfirmDelete
+    }) => eventToDelete => {
+      setConfirmDelete(eventToDelete);
+    }
```

Comment:
Ska vi försöka hålla isär crud, form och state-specifika containers?
Tänker typ:
```js
const EventCRUD = compose(
  getContext({ firebase: PropTypes.object }),
  withHandlers({
    // TODO: move handlers to store/events/crud
    onCreate: ({ firebase, fields }) => () => {
      const ref = firebase.database().ref().child("events").push({
        ...fields,
        owner: firebase.auth().currentUser.uid
      });
      firebase.geo.set(ref.key, [fields.lat, fields.lng]);
    },
    onUpdate: ({ firebase }) => event =>
      firebase.database().ref(`events/${event.id}`).update(event),
    onDelete: ({ firebase }) => event =>
      firebase.database().ref(`events/${event.id}`).remove(),
  })
);
```
Eller kanske till och med:
```js
const handlers = {
  events: require("store/events/crud"),
  users: require("store/users/crud"),
}
const createCRUD = resource =>
  compose(
    getContext({ firebase: PropTypes.object }),
    withHandlers(handlers[resource])
  );
```
Vore nice att hålla containers så atomiska som möjligt, alltså bara göra en grej och sen istället kombinera flera genom compose, typ:
```js
compose(
  eventForm,
  crud("events")
)(EventDetails)`
```
Skulle vi kunna bli av med visibility staten på nåt sätt? Kanske delegera till routern? Typ:
`${resource}/${event}/${action}`
`event/eventId/edit`
`event/eventId/attend`
`user/eventId/attend`

De når vi sen genom `props.match.params.resouce/id/action`